flang/lib/Semantics/resolve-names.cpp

```diff
@@ -6650,7 +6650,7 @@ bool DeclarationVisitor::Pre(const parser::BasedPointer &) {
 
 void DeclarationVisitor::Post(const parser::BasedPointer &bp) {
   const parser::ObjectName &pointerName{std::get<0>(bp.t)};
-  auto *pointer{FindSymbol(pointerName)};
+  auto *pointer{FindInScope(pointerName)};
```

Comment:
Do you need a test that demonstrates that an attempt to host-associate a Cray pointer will now fail with `implicit none (type)`?

Reply:
I think the newly added test (resolve125.f90) shows what you are looking for, but let me know if you where thinking of something else.